sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -56,18 +58,36 @@ process.on("exit", (code: number) => {
 
 const providerCache: { [key: string]: dynamic.ResourceProvider } = {};
```

Comment:
Not strictly in this PR's diff, but I wonder if `Record<string, dynamic.ResourceProvider>` would work here.